registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/validator/LostRidRequestValidator.java

```diff
@@ -0,0 +1,205 @@
+package io.mosip.registration.processor.status.validator;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.TimeZone;
+
+import org.joda.time.DateTime;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.env.Environment;
+import org.springframework.format.datetime.joda.DateTimeFormatterFactory;
+import org.springframework.stereotype.Component;
+
+import io.mosip.kernel.core.exception.ExceptionUtils;
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.registration.processor.core.constant.LoggerFileConstant;
+import io.mosip.registration.processor.core.exception.WorkFlowSearchException;
+import io.mosip.registration.processor.core.exception.util.PlatformErrorMessages;
+import io.mosip.registration.processor.core.logger.RegProcessorLogger;
+import io.mosip.registration.processor.core.workflow.dto.SortInfo;
+import io.mosip.registration.processor.status.dto.FilterInfo;
+import io.mosip.registration.processor.status.dto.LostRidRequestDto;
+import io.mosip.registration.processor.status.exception.LostRidValidationException;
+import io.mosip.registration.processor.status.exception.RegStatusAppException;
+import io.mosip.registration.processor.status.exception.RegStatusValidationException;
+
+@Component
+public class LostRidRequestValidator {
+
+	/** The Constant DATETIME_TIMEZONE. */
+	private static final String DATETIME_TIMEZONE = "mosip.registration.processor.timezone";
+
+	/** The Constant DATETIME_PATTERN. */
+	private static final String DATETIME_PATTERN = "mosip.registration.processor.datetime.pattern";
+
+	/** The mosip logger. */
+	Logger regProcLogger = RegProcessorLogger.getLogger(RegistrationStatusRequestValidator.class);
+
+	/** The Constant REG_STATUS_SERVICE. */
+	private static final String REG_STATUS_SERVICE = "RegStatusService";
+
+	/** The Constant REG_STATUS_APPLICATION_VERSION. */
+	private static final String REG_LOSTRID_APPLICATION_VERSION = "mosip.registration.processor.lostrid.version";
+
+	private static final String REG_LOSTRID_SERVICE_ID = "mosip.registration.processor.lostrid.id";
+
+	/** The env. */
+	@Autowired
+	private Environment env;
+
+	/** The grace period. */
+	@Value("${mosip.registration.processor.grace.period}")
+	private int gracePeriod;
+
+
+	/**
+	 * Validate.
+	 *
+	 * @param registrationStatusRequestDTO the registration status request DTO
+	 * @param serviceId                    the service id
+	 * @throws RegStatusAppException the reg status app exception
+	 */
+	public void validate(LostRidRequestDto lostRidRequestDto)
+			throws RegStatusAppException, WorkFlowSearchException {
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(), "",
```

Comment:
Dont use session id and registration id in logs, use the {} {} type of logs. Take care of this everywhere

Reply:
taken care

Reply:
After the change, did you verify the logs for the correct logs ?

Reply:
yes

Reply:
regProcLogger.debug("{}", "{}", "", "LostRidRequestValidator::validate()::entry");

Above has to be changed to 

regProcLogger.debug("LostRidRequestValidator::validate()::entry");

Reply:
taken care